dataikuapi/dss/ml.py

```diff
@@ -315,43 +319,951 @@ def save(self):
+            return self.__class__.__name__ + "(hyperparameter=\"{}\", value={}, accepted_values={})".format(self.name,
+                                                                                                            self._algo_settings[self.name],
+                                                                                                            self.accepted_values)
+        else:
+            return self.__class__.__name__ + "(hyperparameter=\"{}\", value={})".format(self.name, self._algo_settings[self.name])
+
+    __str__ = __repr__
+
+    _pretty_repr = __repr__
+
+    def set_value(self, value):
+        """
+        :param str value:
+        :return: current SingleValueHyperparameterSettings
+        """
+        if self.accepted_values is not None:
+            assert value in self.accepted_values, "Invalid value for hyperparameter {}. Must be in {}".format(self.name, json.dumps(self.accepted_values))
+        self._algo_settings[self.name] = value
+        return self
+
+
+class PredictionAlgorithmSettings(dict):
+
+    """
+    Object to read and modify the settings of a prediction ML algorithm.
+
+    Do not create this object directly, use :meth:`DSSMLTask.get_algorithm_settings(algorithm)` instead
+    """
+    def __init__(self, raw_settings, hyperparameter_search_params):
+        super(PredictionAlgorithmSettings, self).__init__(raw_settings)
+        self._hyperparameter_search_params = hyperparameter_search_params
+        self._hyperparameters_registry = dict()
+
+    def _register_numerical_hyperparameter(self, name):
+        self._hyperparameters_registry[name] = NumericalHyperparameterSettings(name, self)
+        return self._hyperparameters_registry[name]
+
+    def _register_categorical_hyperparameter(self, name):
+        self._hyperparameters_registry[name] = CategoricalHyperparameterSettings(name, self)
+        return self._hyperparameters_registry[name]
+
+    def _register_single_category_hyperparameter(self, name, accepted_values=None):
+        self._hyperparameters_registry[name] = SingleCategoryHyperparameterSettings(name, self, accepted_values=accepted_values)
+        return self._hyperparameters_registry[name]
+
+    def _register_single_value_hyperparameter(self, name, accepted_types=None):
+        self._hyperparameters_registry[name] = SingleValueHyperparameterSettings(name, self, accepted_types=accepted_types)
+        return self._hyperparameters_registry[name]
+
+    def _repr_html_(self):
+        res = "<pre>" + self.__class__.__name__ + "\n"
+        for name, hyperparam_settings in self._hyperparameters_registry.items():
+            res += "\"{}\": {}".format(name, hyperparam_settings._pretty_repr()) + "\n"
+        res += "</pre>"
+        return res + "<details><pre>{}</pre></details>".format(self.__repr__())
+
+    def __repr__(self):
+        return self.__class__.__name__ + "(values={})".format(super(PredictionAlgorithmSettings, self).copy())
```

Comment:
Why do we need to call the super method here, as it's not overriden in the class ?